src/Microsoft.ML.Data/Scorers/BinaryClassifierScorer.cs

```diff
@@ -197,14 +197,32 @@ private protected override void SaveAsOnnxCore(OnnxContext ctx)
             for (int iinfo = 0; iinfo < Bindings.InfoCount; ++iinfo)
                 outColumnNames[iinfo] = Bindings.GetColumnName(Bindings.MapIinfoToCol(iinfo));
 
-            //Check if "Probability" column was generated by the base class, only then
-            //label can be predicted.
+            /* If the probability column was generated, then the classification threshold is set to 0.5. Otherwise,
+               the predicted label is based on the sign of the score.
+               REVIEW: Binarizer should always have at least two output columns?
+             */
+            string opType = "Binarizer";
+            var binarizerOutput = ctx.AddIntermediateVariable(null, "BinarizerOutput", true);
+
             if (Bindings.InfoCount >= 3 && ctx.ContainsColumn(outColumnNames[2]))
             {
-                string opType = "Binarizer";
-                var node = ctx.CreateNode(opType, new[] { ctx.GetVariableName(outColumnNames[2]) },
-                    new[] { ctx.GetVariableName(outColumnNames[0]) }, ctx.GetNodeName(opType));
+                var node = ctx.CreateNode(opType, ctx.GetVariableName(outColumnNames[2]), binarizerOutput, ctx.GetNodeName(opType));
                 node.AddAttribute("threshold", 0.5);
+
+                opType = "Cast";
+                node = ctx.CreateNode(opType, binarizerOutput, ctx.GetVariableName(outColumnNames[0]), ctx.GetNodeName(opType), "");
+                var t = InternalDataKindExtensions.ToInternalDataKind(DataKind.Boolean).ToType();
+                node.AddAttribute("to", t);
+            }
```

Comment:
This section seems to be common to both the if and else block. Can this be factored out?